taskvine/src/worker/vine_transfer_server.c

```diff
@@ -96,6 +97,7 @@ void vine_transfer_server_stop()
 
 void vine_transfer_server_address( char *addr, int *port )
 {
+	domain_name_cache_guess(addr);
```

Comment:
This is the problem: `domain_name_cache_guess` fills in its argument with the guessed domain name.
`addr` is a string only long enough to hold an IP address.